src/languages/java.rs

```diff
@@ -68,59 +104,51 @@ impl JavaProcessor {
         None
     }
 
-    /// Extracts the container (class or interface) with the method declaration and body.
-    ///
-    /// @example
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    ///
-    /// is split into:
-    ///
-    /// ```java
-    /// public class MyClass {
-    ///     public void method() {
-    /// ```
-    /// and
-    ///
-    /// ```java
-    ///         methodBody();
-    ///     }
-    /// }
-    /// ```
-    fn extract_container_with_method(
+    /// Extracts the method with its full context from nested classes/interfaces
+    fn extract_method_with_context(
         &self,
-        content: &str,
-        container_node: Node,
         method_node: Node,
+        content: &str,
+        stack: &[(Node, String)],
     ) -> (String, String) {
-        // Find the container declaration start (including JavaDoc/comments before it)
-        let container_start = self.find_container_start_with_comments(container_node);
-        let container_body_start = self.find_container_body_start(container_node);
+        let mut declaration = String::new();
 
-        if let Some(body_start) = container_body_start {
-            let mut declaration = String::new();
+        // Build the full context from the stack
+        for (i, (container_node, _container_name)) in stack.iter().enumerate() {
+            if i > 0 {
+                declaration.push('\n');
+            }
 
-            // Add everything from the container start (including JavaDoc) up to the opening brace
-            let container_decl = &content[container_start..body_start];
-            declaration.push_str(container_decl);
-            declaration.push('\n');
+            // Add container declaration with proper indentation
+            let container_start = self.find_container_start_with_comments(*container_node);
+            let container_body_start = self.find_container_body_start(*container_node);
 
-            // Split method into declaration and body using AST
-            let (method_declaration, method_body) =
-                self.split_method_declaration_and_body(method_node, content);
+            if let Some(body_start) = container_body_start {
+                let container_decl = &content[container_start..body_start + 1];
+                // Use the original indentation from the source - only adjust if this is a nested container
+                if i == 0 {
+                    // First container - use original indentation as-is
+                    declaration.push_str(container_decl);
```

Comment:
What if the outermost container (class) is for some reason indented as well? We don't want to fuck up the indentation, so it is best to take into account its indentation as well. Moreover, we should not be taking the indentation of the container itself here (of a class), but rather of the earliest node that we include in the declaration (which in this case might be a javadoc comment). In order to do that we need to also change the 'find container start with comment' method so that it returns the starting node (so a javadoc comment node, or a class/interface/method declaration node) and the code that uses this method should be able to get both the absolute node position from it and the indentation